### PR TITLE
Added full bibliography support, appendices and an IHEID ggplot template

### LINKED ISSUE #26: add cite_packages() function
https://github.com/easystats/report/blob/master/R/cite_packages.R

### LINKED ISSUE #32: Change the placeholder text
When printing just a single chapter with `chapter_pdf()`, the section titles of all the other chapters are included in addition and some 'placeholder' text is entered under some of these extra section titles. These extra section titles either should not appear or appear only in the table of contents printed at the start of the chapter preview.

### LINKED ISSUE #39: iheiddown ggplot theme

### LINKED ISSUE #42: Chapter by chapter preview outputs titles of unpreviewed sections/chapters
When printing just a single chapter with `chapter_pdf()`, the section titles of all the other chapters are included in addition and some 'placeholder' text is entered under some of these extra section titles. These extra section titles either should not appear or appear only in the table of contents printed at the start of the chapter preview.

### Links:
The issue seems to stem from the bookdown function *preview_chapter()* [Doc Here](https://bookdown.org/yihui/bookdown/preview-a-chapter.html). The documentation, as well as previous issues, have mentionned it on Github and the response from Yi Hui is that this is done by design. Also see, [this issue](https://github.com/rstudio/bookdown/issues/497) and [this issue](https://community.rstudio.com/t/problems-previewing-chapter-in-bookdown/3515)

### LINKED ISSUE #45: Revising the formatting of the bibliography
Currently, the bibliography displays with the same style as the rest of the document instead of having a neater "bibliography like" indentation and overall style.

### LINKED ISSUE #46: Create an appendix Rmd file
Create an appendix Rmd file to dump tables, graphs, and more tables that would otherwise clutter the main corpus of the thesis.

### LINKED ISSUE #47: Page numbers do not display when "draft option" is selected
### Description:

Page numbers at the bottom of the pages in the chapters do not display if the draft option is selected. This should be addressed for ease of communication between parties when correcting the draft of the thesis.

### LINKED ISSUE #49: Issue with the multiline thesis title on macOS
**Describe the bug**
On macOS, the thesis title displays as "Conclusion" instead of the multiline title defined in the YAML header macro. We have managed to alleviate the issue by replacing the name of the YAML macro by something else than "title". 

**Desktop:**
 - OS:  MacOS Big Sur 11.1
 - R Version: 4.0.3
 - RStudio Version 1.4

Solved by renaming the "title macro" to "doctitle" in the Yaml header.

### LINKED ISSUE #50: Test theme_iheid() output
- https://github.com/r-lib/vdiffr

- https://stackoverflow.com/questions/31038709/how-to-write-a-test-for-a-ggplot-plot

### PR DESCRIPTION
**Pull request by BBieri**
# Description

## Package

- Updated README to inform users that new theses must be named "00-ProjectTitle" to avoid issues when merging Rmd files together
- Closed #39 by adding the `{ggplot2}` theme `theme_iheid()`
  - Fixed #50 by adding tests for `theme_iheid()`

## Thesis

- Closed #32 by changing lorem ipsum to LOTR references
- Fixed #49 by changing the "title" macro to "doctitle" to avoid an issue with Pandoc conversion to Latex on MacOS
- Fixed #47 by tweaking the draft macro to display page numbers properly when activated
- Fixed #42 where the `chapter_pdf()` function displayed placeholder text and headers for other unpreviewed chapters when knitting
- Closed #26 by adding an automatic package citations function to the thesis template.
  - Fixed #45 by changing the bibliography management tool to `{biblatex}` and implemented a series of macros in the YAML header of the index file
- Closed #46 by adding an appendix example that can include all the code from chapter code chunks

## Syllabus

- Closed #32 by changing lorem ipsum to LOTR references. 

# Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] The package builds on my OS without issues (please add workstation details)
- [x] My changes generate no new warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (.R, NEWS.md)
- [x] I have bumped the version by the appropriate increment in the DESCRIPTION file (major, minor, patch)

### CODE REVIEW

File: inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd [hunk 1]
```diff
@@ -45,6 +45,7 @@ bibliography: [bib/packages.bib,bib/references.bib]
 bibliography-heading-in-pdf: References
 bib-humanities: true   #set to true if you want in-text references formatted as author-year
 doi-in-bibliography: true #set to true if you want DOI's to be shown in the bibliography
+# csl: apa.csl #Uncomment to integrate the bibliography style file that you have put in the root folder of your thesis. Search the Zotero Stlye Repository for the required citation style (https://www.zotero.org/styles).
```
Comment (jhollway) [inline inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd:48]: This is great. Does it work via pandoc or the LaTeX template?

Let's preselect one intext citation style and one footnote citation style as options for different departments for now. Can you create a shortlist?

**Review by jhollway — CHANGES_REQUESTED** [review]

File: inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd (hunk 1 above)
Comment (BBieri) [reply inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd]: It works via Pandoc. The user can specify the desired citation style by referencing a csl file in the bibstyles folder (just updated it, it's no longer the root directory) in the YAML header of the index file. 

I preselected the following bibliography styles and downloaded their latest version in the bibstyles file. 
- APA (for social sciences)
- MLA (for humanities)
- Chicago author-date (for social sciences)
- Chicago full note (for history)

Let me know if there is another citation style I should include by default.

File: NAMESPACE [hunk 2]
```diff
@@ -7,10 +7,12 @@ export(chapter_pdf)
 export(iheid_palette)
 export(iheid_palettes)
 export(syllabus_pdf)
+export(theme_iheid)
```
Comment (jhollway) [inline NAMESPACE:10]: Can you please document this function addition in NEWS & PR description

File: NEWS.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+## Package
+
+* Fixed #42 where the `chapter_pdf()` function displayed placeholder text and headers for other unpreviewed chapters when knitting. 
+* Closed #26 by adding an automatic package citations function to the thesis template.
+* Updated the README to signal that new projects must be named "00-ProjectTitle" to avoid issues when merging Rmd files together.
+
+## Thesis
+
+* Closed #32 by changing Harry Potter references to LOTR references. 
+
```
Comment (jhollway) [inline NEWS.md:12]: Document `theme_iheid()`

File: R/theme.R [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+#' 
+#' @importFrom ggplot2 theme
+#' @examples 
+#' \dontrun{
+#' ggplot(iris, aes(x = Sepal.Length, y = Sepal.Width)) + 
+#' geom_point() + ggtitle("Example Plot") + theme_iheid()
+#' }
+#' @export
+theme_iheid <- function(){
+  font <- "Helvetica"
```
Comment (jhollway) [inline R/theme.R:11]: Or Helvetica Neue?

File: R/theme.R [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  
+  ggplot2::theme(
+    
+    #Text format:
+    #This sets the font, size, type and colour of text for the chart's title
+    plot.title = ggplot2::element_text(family=font,
+                                       size=28,
+                                       face="bold",
+                                       color=iheid_palette("IHEID")[1]),
```
Comment (jhollway) [inline R/theme.R:20]: Named element index, e.g. `iheid_palette("IHEID)["red"]`, easier to read/amend?

File: R/theme.R [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+#' }
+#' @export
+theme_iheid <- function(){
+  font <- "Helvetica"
+  
+  ggplot2::theme(
+    
+    #Text format:
+    #This sets the font, size, type and colour of text for the chart's title
+    plot.title = ggplot2::element_text(family=font,
```
Comment (jhollway) [inline R/theme.R:17]: Need to declare these additional ggplot2 @importFrom?

File: README.md [hunk 7]
```diff
@@ -81,7 +82,8 @@ It's very easy to get started.
 1. Click the new document symbol at the very top left and choose `R Markdown...`. 
 1. Then from the options on the left `From Template`. 
 1. Scroll down through the options on the right until you find `Thesis   {iheiddown}`.
-1. Name your thesis project (if in doubt 'Untitled' is fine) and select where the project should be saved on your harddrive.
+1. Name your thesis project by using the following format: 00-NameOfYourProject (for example: 00-Untitled).
```
Comment (jhollway) [inline README.md:85]: Say `00-MyThesisTitle` rather than `00-Untitled`... more indicative

File: inst/rmarkdown/templates/syllabus/skeleton/skeleton.Rmd [hunk 8]
```diff
@@ -20,28 +20,28 @@ program: |
...
 prof-office-hours: "[Fridays, 17--19](https://bit.ly/goog.ly)"
 
 ####################
 ## Assistant      ##
 ####################
-ta: Neville Longbottom
-ta-email: neville.longbottom@graduateinstitute.ch
+ta: Frodo Baggins
+ta-email: frodo.baggins@graduateinstitute.ch
 ta-office-hours: By appointment
```
Comment (jhollway) [inline inst/rmarkdown/templates/syllabus/skeleton/skeleton.Rmd:33]: This is great, well done @BBieri !

File: inst/rmarkdown/templates/syllabus/skeleton/skeleton.Rmd [hunk 9]
```diff
@@ -86,20 +86,20 @@ draft: true
...
 
-##  `r advdate(startdate, 4)`: Match to needle
+##  `r advdate(startdate, 4)`: Meeting Strider
 
-##  `r advdate(startdate, 5)`: Switching spell
+##  `r advdate(startdate, 5)`: Arrival in Rivendell
 
-##  `r advdate(startdate, 6)`: Mice to snuffboxes
\ No newline at end of file
+##  `r advdate(startdate, 6)`: On the road to Mordor
```
Comment (jhollway) [inline inst/rmarkdown/templates/syllabus/skeleton/skeleton.Rmd]: Nice!

File: NEWS.md [hunk 10]
```diff
@@ -1,3 +1,15 @@
...
+## Package
+
+* Fixed #42 where the `chapter_pdf()` function displayed placeholder text and headers for other unpreviewed chapters when knitting. 
+* Closed #26 by adding an automatic package citations function to the thesis template.
+* Updated the README to signal that new projects must be named "00-ProjectTitle" to avoid issues when merging Rmd files together.
+
+## Thesis
+
+* Closed #32 by changing Harry Potter references to LOTR references.
```
Comment (jhollway) [inline NEWS.md:11]: Maybe highlight that these have been changed in both Thesis and Syllabus outputs?

File: inst/rmarkdown/templates/thesis/skeleton/98-appendices.Rmd [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+```
+
+The goal of this appendix is to echo the code you used in your thesis for a greater sense of transparency and replicability of your research. Note that `ref.labels` can be set to any label. Hence, you can filter the code you want replicated in the appendix by setting labels to the desired code chunks in the various chapters. See [this excellent resource](https://bookdown.org/yihui/rmarkdown-cookbook/code-appendix.html) for more information.
+
+
+This might be particularly useful when you perform model selection to output intermediary steps here instead of in the code to avoid cluttering your report. 
+
+```{r get-labels, echo = FALSE}
+labs = knitr::all_labels()
+labs = setdiff(labs, c("setup", "get-labels"))
```
Comment (jhollway) [inline inst/rmarkdown/templates/thesis/skeleton/98-appendices.Rmd:14]: r assignment should use `<-`

File: inst/rmarkdown/templates/thesis/skeleton/98-appendices.Rmd [hunk 12]
```diff
@@ -0,0 +1,22 @@
+# Appendix: The Echoes of the Code{-}
+
+```{=latex}
+\markboth{Appendix}{Appendix}
+```
+
+The goal of this appendix is to echo the code you used in your thesis for a greater sense of transparency and replicability of your research. Note that `ref.labels` can be set to any label. Hence, you can filter the code you want replicated in the appendix by setting labels to the desired code chunks in the various chapters. See [this excellent resource](https://bookdown.org/yihui/rmarkdown-cookbook/code-appendix.html) for more information.
```
Comment (jhollway) [inline inst/rmarkdown/templates/thesis/skeleton/98-appendices.Rmd:7]: Separate into separate lines?

File: inst/rmarkdown/templates/thesis/skeleton/99-references.Rmd [hunk 13]
```diff
@@ -1,7 +1,11 @@
...
 You don't really need to alter this chapter unless you want to change the name of the bibliography section 
 to 'Bibliography', 'Literature', or something similar.
 Any references included in the chapters will be printed here automatically.
--->
\ No newline at end of file
+-->
+```{=latex}
+\markboth{References}{References}
+```
+<!-- This quiet code chunk serves as a custom style for the reference section and adds a custom header Don't change it unless you really want to. -->
```
Comment (jhollway) [inline inst/rmarkdown/templates/thesis/skeleton/99-references.Rmd:11]: Missing full stop or some other punctuation between "header" and "Don't"

File: inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd [hunk 14]
```diff
@@ -41,10 +41,16 @@ mini-lof: false  # mini-list of figures by start of each chapter?
 ####################
 ## Bibliography   ##
 ####################
-bibliography: bib/references.bib
+bibliography: bib/references.bib #Set your bibliography file here.
+packagebibliography: bib/packages.bib #Sets the automatically generated R-Package Bibliography.
```
Comment (jhollway) [inline inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd:45]: If this is done automatically, does it have to be listed here in the skeleton?

File: inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd [hunk 15]
```diff
@@ -41,10 +41,16 @@ mini-lof: false  # mini-list of figures by start of each chapter?
...
 bibliography-heading-in-pdf: References
-bib-humanities: true   #set to true if you want in-text references formatted as author-year
-doi-in-bibliography: true #set to true if you want DOI's to be shown in the bibliography
+bib-style: authoryear # See https://www.overleaf.com/learn/latex/biblatex_citation_styles for a list of the commonly used built-in citations styles of biblatex.
+bib-sorting: nyt #See https://www.overleaf.com/learn/latex/Articles/Getting_started_with_BibLaTeX for different bibliography sorting options.
+doi-in-bibliography: true #set to true if you want DOI's to be shown in the bibliography.
+bib-isbn: true #Set to true if you want the ISBN to be displayed
+bib-dashed: false #Set to true if you want dashed entries instead of repeaded author names if your bibliography contains two entries of the same author.
+bib-smallfont: false #Set to true if you want a small fontsize for your bibliography.
+citeall: true #Set this to true if you want all elements in your .bib file to appear in your bibliography (i.e. all R-packages you used).
```
Comment (jhollway) [inline inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd:53]: Too many options? Keep it simple by not listing these as options, and checking for them in the pandoc/latex template. If they are not there, we use sensible defaults.

File: inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd [hunk 16]
```diff
@@ -73,6 +79,12 @@ The {.unnumbered} option here means that the introduction will be
 "Chapter 0." You can also use {-} for no numbers on chapters.
 -->
 
+```{r, include=FALSE}
+knitr::write_bib(c(.packages(), "bookdown"), "bib/packages.bib")
+```
+
+<!-- This chunk creates automatically a .bib file to cite the loaded packages in the bibliography. Do not forget to uncomment "nocite: '@*'" in the header if you want all packages to be cited even if you did not mention them in the body of the text.-->
```
Comment (jhollway) [inline inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd:86]: Put the instructions above or within so that users don't accidentally delete or get turned off as they read from top to bottom

File: inst/rmarkdown/templates/thesis_pdf/resources/template.tex [hunk 17]
```diff
@@ -30,9 +30,10 @@
 
 \fancypagestyle{plain}{\fancyhf{}\fancyfoot[C]{\emph{\thepage}}}
 
-% This adds a "DRAFT" footer to every normal page.  (The first page of each chapter is not a "normal" page.)
+% This adds a "DRAFT" footer to every normal page.  (The first page of each chapter is not a "normal" page.) Page numbers are also printed on the right side.
 $if(draft)$
 \fancyfoot[C]{\emph{DRAFT Printed on \today}}
```
Comment (jhollway) [inline inst/rmarkdown/templates/thesis_pdf/resources/template.tex:35]: Hmm, I don't like this anymore. Can you please make it `DRAFT: \today`?

**Review by jhollway — CHANGES_REQUESTED** [review]

**Comment by jhollway** [comment]
Please also keep the NEWS and PR description in sync

File: inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd (hunk 14 above)
Comment (BBieri) [reply inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd:45]: True, I can define it directly in the tex template but if the user wants to change the file path of the bibliography, this would give him/her a greater level of flexibility.

File: inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd (hunk 15 above)
Comment (BBieri) [reply inst/rmarkdown/templates/thesis/skeleton/skeleton.Rmd:53]: I don't necessarily agree on this one in the sense that the `{iheiddown}` package is aimed at people who do quantitative analyses and in my opinion value the customizability of the template. It's a bit like the macOS vs Windows or Linux debate; do you prefer something that works out of the box or do you prefer something that you can easily customize?

File: R/theme.R (hunk 4 above)
Comment (BBieri) [reply R/theme.R:11]: Helvetica and Helvetica Neue are fonts that are available natively only on macOS and their equivalent is Arial on Windows machines. I'll try to find a workaround to set the font depending on the OS the user is running otherwise we might have to settle for a different font available across all platforms.

Comment (BBieri) [reply R/theme.R:11]: I would recommend leaving it at "sans", the default option which works across platforms and works with base R. Users who would want to customize the fonts (and use Comic Sans MS for example) can on their own add other Windows and MacOS fonts by using the `{extrafont}`. More details [here](https://cran.r-project.org/web/packages/extrafont/README.html).

**Review by jhollway — CHANGES_REQUESTED** [review]

File: R/theme.R (hunk 4 above)
Comment (BBieri) [reply R/theme.R:11]: Changed Helvetica to "sans" for the sake of interoperability. I will mention the option to add local fonts with`{extrafonts}` in the upcoming vignette.

File: R/theme.R (hunk 5 above)
Comment (BBieri) [reply R/theme.R:20]: Changed it to `IHEIDRed`.

File: R/theme.R (hunk 6 above)
Comment (BBieri) [reply R/theme.R:17]: Not sure I understood you here. Could you rephrase the comment?

Comment (jhollway) [reply R/theme.R:17]: Quicker to just do it. Done in last commit.

Comment (BBieri) [reply R/theme.R:17]: Thanks, just saw it and read the Roxygen doc. Everything is much clearer now.

**Review by jhollway — APPROVED** [review]
